sdk/tests/core/inc/az_test_precondition.h

```diff
@@ -4,9 +4,12 @@
 #ifndef _az_TEST_PRECONDITION_H
 #define _az_TEST_PRECONDITION_H
 
-#include <assert.h>
+// These headers must be included prior to including cmocka.h.
 #include <setjmp.h>
 #include <stdarg.h>
+#include <stddef.h>
+
```

Comment:
I don't think we need this specific empty line. All are headers from standard library, but in adjacent include blocks, there's probably no reason.

Reply:
I wanted to make it clear what headers specifically need to be included before cmocka.h and hence the comment.

We have this type of grouping in a bunch of places. Merging them together causes VS auto-format to re-order them, which I don't want since commenting the three headers becomes more cumbersome. That's why I split them up.

If you feel strongly about it, please feel free to fix it up across the repo and let's be consistent. For this PR, I am going to limit the change and leave it as is. It doesn't have a material impact on correctness, so low priority, imo.

Reply:
If so, then sure, it is exactly a valid reason to create a group of includes, the change is not needed in this case.
(I know that grouping would lead to sorting).